CMakeLists.txt

```diff
@@ -0,0 +1,1141 @@
+#
```

Comment:
I suspect that there will be very few fans, if any, on the Git mailing list, of copying the CMake definition. Not only does this willfully risk diverging `CMakeLists.txt` files, it is also highly unclear why this is needed if the stated goal is to apply clang-tidy to the Git project.

Speaking of that goal: Patch series that touch so many files at once are likely to cause a lot of conflicts with other patch series that are in flight, and such conflicts rarely remain code conflicts and evolve into full-blown social conflicts. You may want to use a much more incremental approach, gaging what files in Git's code base are currently untouched by actively-developed patch series and handling only those in a first step.